lib/database/database_helper.dart

```diff
@@ -0,0 +1,152 @@
+import 'dart:async';
+import 'dart:io' as io;
+
+import 'package:time_tracker/models/task.dart';
+import 'package:path/path.dart';
+import 'package:path_provider/path_provider.dart';
+import 'package:sqflite/sqflite.dart';
+import 'package:flutter/widgets.dart';
+
+class DatabaseHelper {
+  bool test;
+
+  static final DatabaseHelper _instance = new DatabaseHelper.internal();
+
+  factory DatabaseHelper({test = false}) {
+    _instance.test = test;
+    return _instance;
+  }
+
+  static Database _db;
+
+  Future<Database> get db async {
+    if (_db != null) return _db;
+    _db = await initDb();
+    return _db;
+  }
+
+  DatabaseHelper.internal();
+
+  initDb() async {
+    WidgetsFlutterBinding.ensureInitialized();
+    // io.Directory documentsDirectory = await getApplicationDocumentsDirectory();
+    // String path = join(documentsDirectory.path, "main.db");
+    // var theDb = await openDatabase(path, version: 1, onCreate: _onCreate);
+    // return theDb;
+    print("making the database");
+
+    Database database = await openDatabase(
+      join(await getDatabasesPath(), test ? 'database_test.db' : 'data.db'),
+      // When the database is first created, create a table to store tasks.
+      onCreate: (db, version) {
+        // Run the CREATE TABLE statement on the database.
+        return db.execute(
+          "CREATE TABLE task(id INTEGER PRIMARY KEY, description TEXT, clock INTEGER, category TEXT)",
+        );
+      },
+      // Set the version. This executes the onCreate function and provides a
+      // path to perform database upgrades and downgrades.
+      version: 1,
+    );
+
+    return database;
+  }
+
+  // get
+  Future<Task> getTask(int id) async {
+    // Get a reference to the database.
+    final Database dbClient = await db;
+
+    // get the Task from the Database.
+    List<Map<String, dynamic>> tasks = await dbClient.query(
+      'task',
+      // Use a `where` clause to delete a specific Task.
+      where: "id = ?",
+      // Pass the Task's id as a whereArg to prevent SQL injection.
+      whereArgs: [id],
+    );
+    if (tasks.length == 0) {
+      return null;
+    }
+
+    return Task(
+        id: tasks[0]["id"],
+        description: tasks[0]["description"],
+        category: tasks[0]["category"],
+        clock: tasks[0]["clock"]);
+  }
+
+  // Define a function that inserts tasks into the database
+  Future<void> insertTask(Task task) async {
+    print("insert");
+    // Get a reference to the database.
+    final Database dbClient = await db;
+
+    // Insert the Task into the correct table. You might also specify the
+    // `conflictAlgorithm` to use in case the same Task is inserted twice.
+    //
+    // In this case, replace any previous data.
+    await dbClient.insert(
```

Comment:
I would suggest changing the conflict algorithm from `replace` to `fail` so that debugging is easier in the future.